src/nomad_catalysis/schema_packages/catalysis.py

```diff
@@ -394,3 +412,1107 @@ def normalize(self, archive, logger):
+                fig0.add_trace(
+                    go.Scatter(
+                        x=x,
+                        y=self.results[0].products[i].selectivity,
+                        name=self.results[0].products[i].name,
+                    )
+                )
+            fig0.update_layout(title_text='Selectivity', showlegend=True)
+            fig0.update_xaxes(title_text=x_text)
+            fig0.update_yaxes(title_text='Selectivity (%)')
+            self.figures.append(
+                PlotlyFigure(label='Selectivity', figure=fig0.to_plotly_json())
+            )
+        fig1 = self.conversion_plot(x, x_text, logger)
+        if fig1 is not None:
+            self.figures.append(
+                PlotlyFigure(label='Conversion', figure=fig1.to_plotly_json())
+            )
+
+        if self.results[0].rates:
+            fig = go.Figure()
+            for i, c in enumerate(self.results[0].rates):
+                fig.add_trace(
+                    go.Scatter(
+                        x=x,
+                        y=self.results[0].rates[i].rate,
+                        name=self.results[0].rates[i].name,
+                    )
+                )
+            fig.update_layout(title_text='Rates', showlegend=True)
+            fig.update_xaxes(title_text=x_text)
+            fig.update_yaxes(title_text='rates (g product/g cat/h)')
+            self.figures.append(
+                PlotlyFigure(label='Rates', figure=fig.to_plotly_json())
+            )
+        if not self.results[0].reactants_conversions:
+            return
+        if self.results[0].reactants_conversions[0].conversion and (
+            self.results[0].products[0].selectivity
+        ):
+            for i, c in enumerate(self.results[0].reactants_conversions):
+                name = self.results[0].reactants_conversions[i].name
+                fig = go.Figure()
+                for j, p in enumerate(self.results[0].products):
+                    fig.add_trace(
+                        go.Scatter(
+                            x=self.results[0].reactants_conversions[i].conversion,
+                            y=self.results[0].products[j].selectivity,
+                            name=self.results[0].products[j].name,
+                            mode='markers',
+                        )
+                    )
+                fig.update_layout(title_text='S-X plot ' + str(i), showlegend=True)
+                fig.update_xaxes(title_text=name + ' Conversion (%)')
+                fig.update_yaxes(title_text='Selectivity (%)')
+                self.figures.append(
+                    PlotlyFigure(
+                        label='S-X plot ' + name + ' Conversion',
+                        figure=fig.to_plotly_json(),
+                    )
+                )
+
+    def normalize_reaction_conditions(
+        self, archive: 'EntryArchive', logger: 'BoundLogger'
+    ) -> None:
+        if self.reaction_conditions is not None:
```

Comment:
You could probably negate this and return. Then you can reduce the indentation of the stuff below one level:
```py
if self.reaction_conditions is None:
    return
reagents = []
...
```